src/components/content-tab-assistant.tsx

```diff
@@ -60,10 +40,31 @@ export const Message = ( { children, isUser, className }: MessageProps ) => (
 			) }
 		>
 			{ typeof children === 'string' ? (
-				<div
-					id="assistant-markdown"
-					dangerouslySetInnerHTML={ { __html: md.render( children ) } }
-				/>
+				<div id="assistant-markdown">
+					<ReactMarkdown
+						components={ {
+							code( { inline, className, children, ...props } ) {
+								const match = /language-(\w+)/.exec( className || '' );
+								return ! inline && match ? (
+									<SyntaxHighlighter
+										style={ xonokai }
+										language={ match[ 1 ] }
+										PreTag="div"
+										{ ...props }
+									>
+										{ String( children ).replace( /\n$/, '' ) }
+									</SyntaxHighlighter>
+								) : (
+									<code className={ className } { ...props }>
+										{ children }
+									</code>
+								);
```

Comment:
This was also an older iteration, and simplified. 

Regarding the CSS scope, however, in https://github.com/Automattic/studio/pull/196/files#r1621840957 I noted that using the `id` was intentional to guard against unintended styles from interfering with the chat messages. Possibly not necessary to use an `id` scope but I also wanted to check if there was a particular reason to use a class that I may not be considering.

Reply:
Ah, I follow now -- I see that each `<Message />` sent would utilize the same `id` (which would cause multiples of the same `id`. -- `className` is more appropriate. I've updated this in https://github.com/Automattic/studio/pull/208.